src/derivative_utils.jl

```diff
@@ -46,36 +46,48 @@ jacobian update function, then it will be called for the update. Otherwise,
 either ForwardDiff or finite difference will be used depending on the
 `jac_config` of the cache.
 """
-function calc_J(integrator, cache::OrdinaryDiffEqConstantCache, is_compos)
-  @unpack t,dt,uprev,u,f,p = integrator
+function calc_J(integrator, cache::OrdinaryDiffEqConstantCache)
+  @unpack t,uprev,f,p = integrator
+  
   if DiffEqBase.has_jac(f)
     J = f.jac(uprev, p, t)
   else
-    cache.uf.t = t
-    cache.uf.p = p
-    J = jacobian(cache.uf,uprev,integrator)
+    @unpack uf = cache
+    uf.t = t
+    uf.p = p
+    J = jacobian(uf,uprev,integrator)
   end
   integrator.destats.njacs += 1
-  is_compos && (integrator.eigen_est = opnorm(J, Inf))
-  return J
+  
+  if integrator.alg isa CompositeAlgorithm
+    integrator.eigen_est = opnorm(J, Inf)
+  end
+
+  J
 end
 
-function calc_J(nlsolver, integrator, cache::OrdinaryDiffEqConstantCache, is_compos)
-  @unpack t,dt,uprev,u,f,p = integrator
+function calc_J(nlsolver, integrator, cache::OrdinaryDiffEqConstantCache)
```

Comment:
why do we still need 2 dispatches here? They do the same thing.

Reply:
Definitely should remove one. It's just such a massive change that I wanted to do the derivative_utils.jl cleanup in a separate PR/commit.